lib/Migration/Version030901Date20230615085512.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Text\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version030901Date20230615085512 extends SimpleMigrationStep {
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options) {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$table = $schema->getTable('text_steps');
+		if (!$table->hasIndex('ts_session')) {
+			$table->addIndex(['session_id'], 'ts_session');
+			return $schema;
+		}
```

Comment:
Doing it live on upgrade, do think it'll take long? For example, we have almost 300k entries in that table.

Reply:
Yes, tough this is how apps currently do it as far as I can tell. There is a internal/deprecated ADD_MISSING_INDEXES_EVENT but I couldn't find any usage of it. Let me see if that works, maybe we can just introduce an OCP event for that then as well.

Reply:
For it can run while the instance is life, one of the one-time repair/migration job types would do it, even without manual intervention.

Reply:
… but if course that would not be very consistent with usual NC ops.

Reply:
Right, I think I'll still go with the event then as this allows admins to run it during off peak hours. The query itself is already faster than before even without the index.

Reply:
Background jobs can also run during off-peak hours =) but yes, i concur it's the best apporach.